src/main/java/co/unruly/control/pair/Pair.java

```diff
@@ -1,73 +1,83 @@
 package co.unruly.control.pair;
 
-import java.util.Objects;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
 import java.util.function.BiFunction;
 import java.util.function.Function;
 
 /**
  * A basic tuple type
  */
-public class Pair<L, R> {
-
-    public final L left;
-    public final R right;
+public record Pair<L, R>(L left, R right) {
```

Comment:
If it was converted to a record, it should be because the only fields were public final, and thus shouldn't cause any issues (assuming property access was changed from A.B to A.B())